priam/src/main/java/com/netflix/priam/backupv2/MetaFileWriter.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.netflix.priam.backupv2;
+
+import com.google.gson.stream.JsonWriter;
+import com.google.inject.Provider;
+import com.netflix.priam.IConfiguration;
+import com.netflix.priam.backup.AbstractBackupPath;
+import com.netflix.priam.backup.IBackupFileSystem;
+import com.netflix.priam.backup.IFileSystemContext;
+import com.netflix.priam.identity.InstanceIdentity;
+import com.netflix.priam.utils.RetryableCallable;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.filefilter.FileFilterUtils;
+import org.apache.commons.io.filefilter.IOFileFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * This class will help in generation of meta.json files. This will encapsulate all the SSTables that were there on the file system.
+ * Created by aagrawal on 6/12/18.
+ */
+public class MetaFileWriter {
+    private static final Logger logger = LoggerFactory.getLogger(MetaFileWriter.class);
+    private final Provider<AbstractBackupPath> pathFactory;
+    private final IBackupFileSystem backupFileSystem;
+
+    private MetaFileInfo metaFileInfo;
+    private JsonWriter jsonWriter;
```

Comment:
You might want to add a comment here on the usage of JsonWriter (Streaming/ not intend to load the JSON in memory), otherwise, it would be tough to justify the usage considering the code complexity to build a JSON and ordering of method calls needed without enough context.